exec/util/file.go

```diff
@@ -23,18 +22,23 @@ import (
 
 const (
 	DefaultDirectoryPermissions os.FileMode = 0755
-	DefaultFilePermissions      os.FileMode = 0755
+	DefaultFilePermissions      os.FileMode = 0644
 )
 
-var (
-	ErrNotDirectory = errors.New("file is not a directory")
-)
+// in-memory representation of a file
+type File struct {
+	Path     string
+	Contents string
+	Mode     os.FileMode
+	Uid      int
+	Gid      int
+}
 
-func WriteFile(filename, contents string) error {
+func (f *File) Write() error {
 	var err error
 
-	dir := filepath.Dir(filename)
-	if err := EnsureDirectoryExists(dir); err != nil {
+	dir := filepath.Dir(f.Path)
+	if err := os.MkdirAll(dir, DefaultDirectoryPermissions); err != nil {
```

Comment:
The previous stuff was unnecessary, MkdirAll() does the right thing:

>     MkdirAll creates a directory named path, along with any necessary
    parents, and returns nil, or else returns an error. The permission bits
    perm are used for all directories that MkdirAll creates. If path is
    already a directory, MkdirAll does nothing and returns nil.